src/raqm.c

```diff
@@ -581,17 +586,19 @@ _raqm_u8_to_u32 (const char *text, size_t len, uint32_t *unicode)
 
   while ((*in_utf8 != '\0') && (in_len < len))
   {
-    in_utf8 = _raqm_get_utf8_codepoint (in_utf8, out_utf32);
+    int codepoint_len;
+    in_utf8 = _raqm_get_utf8_codepoint (in_utf8, out_utf32, &codepoint_len);
     ++out_utf32;
-    ++in_len;
+    in_len += codepoint_len;
```

Comment:
I think `codepoint_len` is redundant, with small modification the call site can determine how many bytes to skip, something like:
```C
const char *out_utf8 = _raqm_get_utf8_codepoint (in_utf8, out_utf32);
in_len += out_utf8 - in_utf8;
in_utf8 = out_utf8;
```

Reply:
Changed to match